test-data/unit/check-classes.test

```diff
@@ -6891,3 +6891,14 @@ class B(A):  # E: Final class __main__.B has abstract attributes "foo"
 [case testUndefinedBaseclassInNestedClass]
 class C:
     class C1(XX): pass  # E: Name "XX" is not defined
+
+
+[case testArgsKwargsInheritance]
+from typing import Any
+
+class A(object):
+    def f(self, *args: Any, **kwargs: Any) -> int: ...
+
+class B(A):
+    def f(self, x: int) -> int: ...
```

Comment:
I understand that this is very handy in some cases, https://github.com/python/mypy/issues/5876 has a handful of them. But, it is important to remember that this is not type-safe.

```python
from typing import Any

class A(object):
    def f(self, *args: Any, **kwargs: Any) -> int: ...

class B(A):
    def f(self, x: int) -> int: ...

def test(arg: A):
    arg.f(x=0, kw1=1, kw2=2)

test(A())  # ok
test(B())  # mypy is ok with that, but raises TypeError in real-life
```

It breaks Liskov's law.

Maybe we can somehow limit the scope of this new rule? For example:
- It is fine for decorator example from https://github.com/python/mypy/issues/5876
- It is fine for `Protocol` check (when protocol defines less-or-equally broad signature): `Procol.method(self, arg: int)` -> is ok with `Impl.method(self, *args, **kwargs)`
- It is fine for callbacks
- Maybe something else
- It is **not** fine for subclassing

Reply:
Yeah, I think the ask is explicitly for something that is not type safe. The check when "protocol defines less-or-equally broad signature" should already work. The decorator example from #5876 is a little interesting; this PR doesn't actually change inference there.

#5876 is probably a top five most popular mypy issue, so ideally something would be done. Maybe the answer is another flag.